diff --git a/src/scan/context.rs b/src/scan/context.rs
index 75672d9..8866c47 100644
--- a/src/scan/context.rs
+++ b/src/scan/context.rs
@@ -86,8 +86,7 @@ impl ManifestFileContext {
 
         for manifest_entry in manifest.entries() {
             let manifest_entry_context = ManifestEntryContext {
-                // TODO: refactor to avoid the expensive ManifestEntry clone
-                manifest_entry: manifest_entry.clone(),
+                manifest_entry: Arc::clone(manifest_entry),
                 expression_evaluator_cache: expression_evaluator_cache.clone(),
                 field_ids: field_ids.clone(),
                 partition_spec_id: manifest_file.partition_spec_id,
